server/test/incident.js

```diff
@@ -21,604 +23,500 @@ const notFound = `/api/v1/red-flag/${notFoundRedFlagId}`;
-        * **********************************
-        */
-        it('Should return 400 status code for type not found', (done) => {
-            chai.request(app)
-                .post(url)
-                .send(invalidType[0])
-                .end((err, res) => {
-                    res.should.have.status(400);
-                    res.body.should.be.a('object');
-                    expect(res.body.status).to.equal(400);
-                    res.body.message.should.be.a('string');
-                    done();
-                });
-        });
-        it('Should return 400 status code for type not string', (done) => {
-            chai.request(app)
-                .post(url)
-                .send(invalidType[1])
-                .end((err, res) => {
-                    res.should.have.status(400);
-                    res.body.should.be.a('object');
-                    expect(res.body.status).to.equal(400);
-                    res.body.message.should.be.a('string');
-                    done();
-                });
-        });
-        it('Should return 400 status code for an empty type', (done) => {
-            chai.request(app)
-                .post(url)
-                .send(invalidType[2])
-                .end((err, res) => {
-                    res.should.have.status(400);
-                    res.body.should.be.a('object');
-                    expect(res.body.status).to.equal(400);
-                    res.body.message.should.be.a('string');
-                    done();
-                });
-        });
-        it('Should return 400 status code for an inValid type character', (done) => {
-            chai.request(app)
-                .post(url)
-                .send(invalidType[3])
-                .end((err, res) => {
-                    res.should.have.status(400);
-                    res.body.should.be.a('object');
-                    expect(res.body.status).to.equal(400);
-                    res.body.message.should.be.a('string');
-                    done();
-                });
-        });
-        it('Should return 400 status code for an inValid type length', (done) => {
-            chai.request(app)
-                .post(url)
-                .send(invalidType[4])
-                .end((err, res) => {
-                    res.should.have.status(400);
-                    res.body.should.be.a('object');
-                    expect(res.body.status).to.equal(400);
-                    res.body.message.should.be.a('string');
-                    done();
-                });
+  describe('Tests for createIncident API', () => {
+    it('Should return 400 status code for undefined incident date', (done) => {
+      chai.request(app)
+        .post(url)
+        .send(inValidIncidentDate[0])
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          expect(res.body.status).to.equal(400);
+          res.body.message.should.be.a('string');
```

Comment:
Please test what the response `message` and not only check that it's a string. This applies to all the test cases below.

Reply:
Acknowledged, still working on it.